src/common/quantile.cu

```diff
@@ -657,7 +657,9 @@ void SketchContainer::MakeCuts(HistogramCuts* p_cuts, bool is_column_split) {
     size_t column_size = std::max(static_cast<size_t>(1ul), this->Column(i).size());
     if (IsCat(h_feature_types, i)) {
       // column_size is the number of unique values in that feature.
-      CheckMaxCat(max_values[i].value, column_size);
+      if (!is_column_split) {
```

Comment:
why not?

Reply:
Hmm there seems to be a bug dealing with missing columns. Trying to fix it.

Reply:
Fixed the underlying issue. Please take another look.